modules/dnn/src/webgpu/include/buffer.hpp

```diff
@@ -0,0 +1,49 @@
+#ifndef OPENCV_DNN_WEBGPU_BUFFER_HPP
+#define OPENCV_DNN_WEBGPU_BUFFER_HPP
+// #ifdef HAVE_WEBGPU
+#include <dawn/webgpu_cpp.h>
+#include <unistd.h>
+#include <memory>
+// #endif
+namespace cv { namespace dnn { namespace webgpu {
+
+class Buffer 
+{
+// #ifdef HAVE_WEBGPU
+public:
+    Buffer(const std::shared_ptr<wgpu::Device> device);
+    Buffer( const std::shared_ptr<wgpu::Device> device, 
+            const void* data, size_t size, 
+            wgpu::BufferUsage usage = wgpu::BufferUsage::Storage);
+
+    ~Buffer() {bufferMapped_->buffer.Release();}
+    wgpu::Buffer * getWebGPUBuffer() 
+    { return & bufferMapped_->buffer; }
+    wgpu::BufferUsage getBufferUsage() { return usage_;}
+    std::shared_ptr<wgpu::CreateBufferMappedResult> getBufferMapped()
+    { return bufferMapped_; }
+    void* getBufferMappedData() { return bufferMapped_->data; }
+    
+    static void BufferMapReadCallback(WGPUBufferMapAsyncStatus status,
+                                   const void* data,
+                                   uint64_t dataLength,
+                                   void* userdata)
+    {
+        static_cast<Buffer*>(userdata)->mappedData = data;
+    }
+    const void* MapReadAsyncAndWait(); 
+    
+private:
+    Buffer();
+    std::shared_ptr<wgpu::Device> device_;
+    std::shared_ptr<wgpu::CreateBufferMappedResult>  bufferMapped_;
```

Comment:
wgpu::Buffer, instead of wgpu::CreateBufferMappedResult, is a better choice.
Buffer object should keep unmapped unless it is explicitly mapped by calling specific API.